tool

```diff
@@ -0,0 +1,275 @@
+# PROGRAMS_$name.
+PROGRAMS=(
+	'gnokey'
+	'gno'
+	'gnodev'
+	'gnoland'
+	'gnoweb'
+	'gnofaucet'
+	'gnomd'
+	'gnokeykc'
+)
+PROGRAMS_gnokey=(
+	# directory of go.mod
+	'.'
+	# location (relative to go.mod dir)
+	'./gno.land/cmd/gnokey'
+)
+PROGRAMS_gno=(
+	'.'
+	'./gnovm/cmd/gno'
+	# extra go build/install flags
+	"-ldflags -X=github.com/gnolang/gno/gnovm/pkg/gnoenv._GNOROOT=$tool_root"
+)
+PROGRAMS_gnodev=(
+	'./contribs/gnodev'
+	'./cmd/gnodev'
+	"-ldflags -X=github.com/gnolang/gno/gnovm/pkg/gnoenv._GNOROOT=$tool_root"
+)
+PROGRAMS_gnoland=(
+	'.'
+	'./gno.land/cmd/gnoland'
+)
+PROGRAMS_gnoweb=(
+	'.'
+	'./gno.land/cmd/gnoweb'
+)
+PROGRAMS_gnofaucet=(
+	'./contribs/gnofaucet'
+	'.'
+)
+PROGRAMS_gnomd=(
+	'./contribs/gnomd'
+	'.'
+)
+PROGRAMS_gnokeykc=(
+	'./contribs/gnokeykc'
+	'.'
+)
+
+green() {
+	printf "\033[0;32m%s\033[0m\n" "$1"
+}
+
+tool() {
+	cmd="$1"
+	if [ $# -ne 0 ]; then
+		shift
+	fi
+	case "$cmd" in
+		install)
+			local flags=""
+			for program_name in "$@"; do
+				if [[ $program_name == -* ]]; then
+					# TODO: not perfect; does not support flag arguments, unless using -flag= syntax.
+					flags="$flags $program_name"
+					continue
+				fi
+				if ! printf '%s\0' "${PROGRAMS[@]}" | grep -F -x -z -- "$program_name" > /dev/null; then
+					echo "program $program_name does not exist; \`./tool help install\` for a list of available programs"
+					continue
+				fi
+
+				local program_chdir=PROGRAMS_$program_name[0]
+				local program_path=PROGRAMS_$program_name[1]
+				local program_build=PROGRAMS_$program_name[2]
+				(
+					# prints command as it executes
+					set -o xtrace
+					go install -C "$tool_root/"${!program_chdir} ${!program_build} ${flags} ${!program_path}
+				)
+				green "+ $program_name installed"
+			done
+			;;
+
+		fmt)
+			if [ $# -eq 0 ]; then
+				tool rundep mvdan.cc/gofumpt -w .
+				# TODO: format gnovm/stdlibs and examples gno files.
+				return 0
+			fi
+			tool rundep mvdan.cc/gofumpt "$@"
+			# TODO: evaluate `goimports` support
+			;;
+
+		lint)
+			if [ $# -eq 0 ]; then
+				# cd to root, so we can lint entire project.
+				cd "$tool_root"
+				tool rundep github.com/golangci/golangci-lint/cmd/golangci-lint run --config "$tool_root/".github/golangci.yml
+				return 0
+			fi
+			tool rundep github.com/golangci/golangci-lint/cmd/golangci-lint "$@"
+			;;
+
+		tidy)
+			local flags=""
+			local executed=0
+			for directory in "$@"; do
+				if [[ $directory == -* ]]; then
+					# TODO: not perfect; does not support flag arguments, unless using -flag= syntax.
+					flags="$flags $directory"
+					continue
+				fi
+				executed=1
+				(
+					# prints command as it executes them
+					set -o xtrace
+					env -C $directory go mod tidy $flags
+				)
+			done
+			if [ $executed -eq 0 ]; then
+				(
+					# prints command as it executes
+					set -o xtrace
+					find . -name "go.mod" -execdir go mod tidy $flags \;
+				)
+			fi
+			;;
```

Comment:
In my opinion, the previous approach was more straightforward and user-friendly. It allowed for easy copy-pasting to manually run a part of the "suggested flow", which is a practice I frequently do.